### ISSUE TITLE
ato-desktop death orphans consumer + dep provider; session record lacks dep contract metadata

### ISSUE THREAD

**Issue by Koh0920** [ato-run/ato#28]
## Summary

When `ato-desktop` exits (SIGTERM, crash, user-quit) while a guest
capsule session is alive, the consumer process and the dep-contract
provider both survive but are reparented to launchd (PPID=1). They
cannot be torn down by `ato app session stop <id>` because the
session record persisted under
`~/.ato/apps/ato-desktop/sessions/ato-desktop-session-<pid>.json`
has no `dependency_contracts` field.

This is the Desktop-side counterpart of #24 (which covers `ato run`
SIGINT teardown). Both share the same upstream root cause —
`DependencyContractGuard::Drop` doesn't run on abnormal exit — but
the Desktop path has the extra problem that even *manual* recovery
via `ato app session stop` is blocked by missing metadata.

## Repro

1. Build `ato-desktop-mcp`:
   ```sh
   cd crates/ato-desktop && cargo build --bin ato-desktop-mcp --release
   ```
2. Launch `ato-desktop` with the freshly-installed CLI override and
   the env vars WasedaP2P needs:
   ```sh
   ATO_DESKTOP_ATO_BIN=$HOME/.cargo/bin/ato \
   PG_PASSWORD=test SECRET_KEY=test \
   nohup "$HOME/Applications/Ato Desktop.app/Contents/MacOS/ato-desktop" \
     > /tmp/desktop.log 2>&1 &
   ```
3. Drive it to launch the capsule via MCP:
   ```sh
   { printf '{\"jsonrpc\":\"2.0\",\"id\":1,\"method\":\"initialize\",\"params\":{}}\n'
     printf '{\"jsonrpc\":\"2.0\",\"id\":2,\"method\":\"tools/call\",\"params\":{\"name\":\"browser_navigate\",\"arguments\":{\"url\":\"capsule://github.com/Koh0920/WasedaP2P\"}}}\n'
   } | crates/ato-desktop/target/release/ato-desktop-mcp
   ```
4. Wait until `curl http://127.0.0.1:8000/docs` returns 200.
5. Kill the desktop:
   ```sh
   kill -TERM \$(pgrep -f 'Ato Desktop.app/Contents/MacOS/ato-desktop')
   ```
6. Observe orphans:
   ```sh
   ps -o pid,ppid,command -p \$(pgrep -f 'WasedaP2P.*uvicorn')
   #   PID  PPID COMMAND
   #  7417     1 .../uvicorn main:app --host 127.0.0.1 --port 8000
   pgrep -af 'postgres -D \$HOME/.ato/state/wasedap2p-backend'
   #  7403 ... postgres -D ...wasedap2p-backend...pgdata -p 49297
   ```
7. Try the supposed recovery path:
   ```sh
   ato app session stop ato-desktop-session-7417
   ```
   The consumer might be stopped (depending on how stop resolves the
   PID), but the dep-contract postgres provider is **never** torn
   down because the session record at
   `~/.ato/apps/ato-desktop/sessions/ato-desktop-session-7417.json`
   has no field naming the provider. `jq '.dependency_contracts // \"missing\"'`
   on that record → `\"missing\"`.

## Observed leaks

After step 5, with HTTP probe still answering 200:
- ato-desktop:        0 process(es)
- WasedaP2P uvicorn:  PID 7417 (PPID=1, orphan)
- ato dep-pg:         PID 7403 (PPID=1, orphan), \`postmaster.pid\` still present in
  \`\$HOME/.ato/state/wasedap2p-backend@.../pgdata/\`
- TCP 127.0.0.1:8000  bound by the orphan
- TCP 127.0.0.1:49297 bound by the orphan

## Root cause

Two coupled gaps in
`crates/ato-cli/src/app_control/session.rs::start_runtime_session`
(introduced in d4fc2662):

1. **Lifecycle binding.** After `setup_session_dependency_contracts`
   succeeds, we call `persist_background_dependency_contracts(&guard)`
   then `guard.detach()`. \"Detach\" is correct for the long-lived
   provider, but the dep-snapshot we persist is **not** written into
   the session record itself — it's stored under a separate
   sessions/locks/<digest>.lock file keyed by manifest digest, not
   by session id. The session-record JSON we write via
   `write_session_record` (used by `ato app session stop` to look up
   what to kill) never references that lock file.
2. **Consumer reparenting.** The consumer (uvicorn for `display_strategy=web_url`)
   is spawned through nacelle as a regular subprocess of `ato app
   session start`, which itself is a transient subprocess of
   ato-desktop. When ato-desktop dies, the consumer is reparented to
   launchd. There is no setpgid/process-group, no prctl-equivalent,
   and no MacOS launchd job that would propagate the parent's death.

## Expected behavior

- Either: ato-desktop's death tears down its session children (process
  group, kqueue EVFILT_PROC NOTE_EXIT watcher in the consumer that
  exits when parent dies, or an explicit shutdown path triggered by
  `webview.rs` pane close).
- And: the session record persisted via `write_session_record` should
  embed the dep-contract snapshot (provider PIDs, port allocations,
  pgdata path) so that `ato app session stop <id>` is sufficient to
  recover **all** state, even if `ato-desktop` itself crashed.

## Notes

- Discovered while verifying the d4fc2662 fix (\"wire dep contracts
  into Desktop session-start path\"). The happy path (launch + run +
  graceful pane-close) works; only the death-during-session path
  leaks.
- This blocks Step 4 / Step 5 of the session-start refactor plan
  (\"full session lifecycle integration so \`ato stop <id>\` tears
  down dep providers\" + \"reuse path fail-closed for old session
  records lacking dep metadata\").
- Related: #24 (same teardown gap in the `ato run` SIGINT path).

**Comment by Koh0920** [comment]
Pushed fix in b4d76239: `fix(ato-cli): persist desktop session dep metadata for recovery`

This change addresses the metadata / manual-recovery half of #28:
- desktop session records now embed dependency-contract provider metadata instead of relying only on the `~/.ato/run-sessions/<id>/graph.json` sidecar
- `ato app session stop <id>` now falls back to the session record when that sidecar is missing, so it can still tear down recorded dep providers after a desktop crash
- added a regression test covering the missing-sidecar recovery path

Validation run:
- `cargo test -p ato-session-core -- --nocapture`
- `cargo test -p ato-cli --lib stop_session_uses_record_dependency_contracts_when_sidecar_is_missing -- --nocapture`

I am intentionally not closing this issue yet. The abnormal-exit lifecycle half is still open: if `ato-desktop` dies mid-session, the consumer process can still be orphaned/reparented instead of being torn down proactively. This commit only makes the persisted session record sufficient for manual recovery.

**Comment by Koh0920** [comment]
Phase 2 (orchestration cleanup) is absorbed into #73. The wrapper's `SessionRecord.dependency_contracts` becoming non-empty in orchestration mode is now an explicit acceptance criterion of #73. This issue stays open until that lands; closing here once #73's unified `LaunchSessionCore` work ships.
